PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -515,6 +521,40 @@ def miniAOD_customizeCommon(process):
     stage2L1Trigger_2017.toModify(process.prefiringweight, DataEra = "2017BtoF")
     run2_L1prefiring.toModify(task, func=lambda t: t.add(process.prefiringweight))
 
+    from PhysicsTools.PatAlgos.producersHeavyIons.heavyIonJetSetup import removeL1FastJetJECs
+    from Configuration.Eras.Modifier_pp_on_AA_2018_cff import pp_on_AA_2018
+    from Configuration.Eras.Modifier_pp_on_PbPb_run3_cff import pp_on_PbPb_run3
+    (pp_on_AA_2018 | pp_on_PbPb_run3).toModify(process.slimmedJets, src = 'selectedPatJets')
+    (pp_on_AA_2018 | pp_on_PbPb_run3).toModify(process.slimmedCaloJets, src = 'akPu4CaloJets')
+
+    (pp_on_AA_2018 | pp_on_PbPb_run3).toModify(process.ak8PFJetsPuppi, src = 'pfNoPileUpJMEHI')
```

Comment:
Now that I'm reading the changes more in depth (not just at the level of syntax, I think I have more questions/comments.

- should we eventually rename `pfNoPileUpJMEHI` to `pfEmptyCollection` and use it so that the code is a bit more readable?
- Is all of the effort in these added modifiers aiming to disable all practical running of everything that leads to slimmedJetsAK8, slimmedJetsPuppi, slimmedMETsPuppi ? Please add clear comments for parts that just need to be disabled rather than actually updated for a proper physics output.

I would still prefer that these toModify are applied right where the modules are defined (even if it means updating the addJetCollection or other relevant JME tool code), not here.

Reply:
Yeah, aside from updated the collections used in the slimming, the rest is just disabling.   I don't see any difficulty in moving all these modifications to the places where the modules are defined.  In that case, we don't need this miniAOD_customizeHeavyIon, right? Are you also referring to the modifications in heavyIonJetSetup.py after the addJetCollection call?  Indeed that might involve some less trivial intervention in the JME tools.

Reply:
>  In that case, we don't need this miniAOD_customizeHeavyIon, right?

right, this customization function would not be needed

> Are you also referring to the modifications in heavyIonJetSetup.py after the addJetCollection call?

I was referring also to the cases of effectively disabling the modules/sequences; these are not a part of calls in heavyIonJetSetup.py

Reply:
I switched to pfEmptyCollection and moved the jet customizations to modules where they are defined.   Why is the input to slimmedJets replaced with a transient jet collection?  It look to be related to b-tagging.  I haven't done disactivated the MET stuff.  It looks like the main consumer is now the deepMET modules, which seem to require a different strategy to disactivate them.  Setting 'max_n_pf = 0' causes the following crash. 

   [4] Calling method for module DeepMETProducer/'deepMETsResolutionTune'
Exception Message:
error while running session: Invalid argument: Input to reshape is a tensor with 8 values, but the requested shape has 36000
	 [[{{node reshape_3/Reshape}}]]
----- End Fatal Exception ------------------------------------------------